chunky/src/java/se/llbit/chunky/renderer/renderdump/DumpFormat.java

```diff
@@ -0,0 +1,212 @@
+/* Copyright (c) 2012-2016 Jesper Öqvist <jesper@llbit.se>
+ *
+ * This file is part of Chunky.
+ *
+ * Chunky is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Chunky is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with Chunky.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package se.llbit.chunky.renderer.renderdump;
+
+import se.llbit.chunky.renderer.scene.Scene;
+import se.llbit.util.TaskTracker;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+abstract class DumpFormat {
+  private static final SortedMap<Integer, DumpFormat> DUMP_FORMATS = new TreeMap<>();
+  public static final DumpFormat CLASSIC_DUMP_FORMAT;
+  public static final DumpFormat DEFAULT_DUMP_FORMAT;
+
+  static {
+    CLASSIC_DUMP_FORMAT = ClassicDumpFormat.INSTANCE;
+    registerDumpFormat(CLASSIC_DUMP_FORMAT);
+    registerDumpFormat(CompressedFloatDumpFormat.INSTANCE);
+    DEFAULT_DUMP_FORMAT = getNewestDumpFormat();
+  }
+
+  public static void registerDumpFormat(DumpFormat format) {
+    DUMP_FORMATS.put(format.getVersion(), format);
+  }
+
+  public static DumpFormat getDumpFormatForVersion(int dumpVersion) throws IOException {
+    DumpFormat format = DUMP_FORMATS.get(dumpVersion);
+    if (format == null) {
+      throw new IOException("Unknown dump format version \"" + dumpVersion + "\"");
+    }
+    return format;
+  }
+
+  public static DumpFormat getNewestDumpFormat() {
+    return DUMP_FORMATS.get(DUMP_FORMATS.lastKey());
+  }
+
+  public abstract int getVersion();
+
+  /**
+   * Used in the calculation of the total loading/saving task size
+   */
+  protected int getHeaderProgressSize(Scene scene) {
+    return 1;
+  }
+
+  /**
+   * Used in the calculation of the total loading/saving task size
+   */
+  protected int getBodyProgressSize(Scene scene) {
+    return 1;
```

Comment:
Typically this will depend on the scene width and height, so the overrides could be removed:

```suggestion
    return scene.width * scene.height;
```

Reply:
This is a remnant from the time in refactoring before adding progress report to the FloatingPointCompressor